app/helpers/application_helper.rb

```diff
@@ -23,6 +23,14 @@ def source_name(source)
     source.name.present? ? source.name : source.aggregation
   end
 
+  ##
+  # Returns frontend path with correctly joining '/'
+  # @param path String (with or without leading slash)
+  # @return String
+  def frontend_path(path = '')
+    Settings.frontend.url.chomp('/') + '/' + path.reverse.chomp('/').reverse
```

Comment:
Not that efficiency is a big issue here, but it would usually be more efficient to do the following, and may express more directly what's happening:  `.sub(/^/+) { $1 }`, such as:
```
irb(main):016:0* "/abc".sub(/^\/+/) { $1 }
=> "abc"
irb(main):017:0> "abc".sub(/^\/+/) { $1 }
=> "abc"
irb(main):018:0> "//abc".sub(/^\/+/) { $1 }
=> "abc"
irb(main):019:0> "//abc/".sub(/^\/+/) { $1 }
=> "abc/"
irb(main):020:0> "//abc/def".sub(/^\/+/) { $1 }
=> "abc/def"
irb(main):021:0> "abc/def".sub(/^\/+/) { $1 }
=> "abc/def"
```